pvlib/pvsystem.py

```diff
@@ -1372,6 +1398,28 @@ def get_iam(self, aoi, iam_model='physical'):
         else:
             raise ValueError(model + ' is not a valid IAM model')
 
+    def dc_ohms_from_percent(self):
+        """
+        Calculates the equivalent resistance of the wires using
+        :py:func:`pvlib.pvsystem.dc_ohms_from_percent`,
+        `self.losses_parameters["dc_ohmic_percent"]`,
+        `self.module_parameters["V_mp_ref"]`,
+        `self.module_parameters["I_mp_ref"]`,
+        `self.modules_per_string`, and `self.strings_per_inverter`.
+        See :py:func:`pvlib.pvsystem.dc_ohms_from_percent` for details.
+        """
+
+        kwargs = _build_kwargs(['dc_ohmic_percent'],
+                               self.array_losses_parameters)
+
+        kwargs.update(_build_kwargs(['V_mp_ref', 'I_mp_ref'],
```

Comment:
I wonder if there is a way to extend this to the other DC module models. If `module_parameters` is taken from the CEC database, it will include `V_mp_ref` and `I_mp_ref`.

The other cases are Pvsyst, SAPM and PVWatts.

For Pvsyst, pvlib provides no parameter database. The pvsyst DC power functions don't require V_mp_ref or I_mp_ref, so the user would need to know that these must be added.

For SAPM, the expected names are `Vmpo` and `Impo`.

For PVWatts, the ohmic losses model can't be applied since PVWatts only deals with power (not voltage and current separately).

`V_mp_ref`, `I_mp_ref` and `dc_ohmic_percent` are required by the downstream method. Maybe raise a ValueError here if any are missing at this step.

In the ModelChain `dc_ohms_from_percent` method, we could check `dc_model` and raise a more informative message how to correct the ValueError.

Reply:
Ok. I need to sort out the relations between the DC models and the module parameters, which have always been a bit confusing, I admit. I would like to make this applicable for CEC, SAPM, and PVsyst models. Sounds like for SAPM, I can add the option to use Vmpo and Impo if those are present in the module parameters. But for PVsyst, it may require a different workflow, or like you say, to raise a ValueError to say that those parameters need to be added. I will look into this to add an update.

Reply:
Do you know if the Pvsyst software names these two quantities? If so, I'd use those names. A user will have to add them to `module_parameters` regardless since there's no data source from which to inherit the names.

Reply:
I added a section that checks from and adds the parameters in sets of either `V_mp_ref`, `I_mp_ref`; `Vmpo`, `Impo`; or `Vmpp`, Impp` (pvsyst-like, from the documentation and pan file parameters). It should raise a value error if these are not found. If this looks like a workable solution, should more tests be added for each case?